plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeFileOperations.java

```diff
@@ -153,9 +154,49 @@ public void testCreateOrReplaceTableAsSelect()
                         .add(new FileOperation(DATA, "no partition", "OutputFile.create"))
                         .build());
 
+        assertFileSystemAccesses(
+                "CREATE OR REPLACE TABLE test_create_or_replace_as_select AS SELECT 2 different_col_name",
+                ImmutableMultiset.<FileOperation>builder()
+                        .add(new FileOperation(STARBURST_EXTENDED_STATS_JSON, "extendeded_stats.json", "InputFile.exists"))
+                        .add(new FileOperation(TRINO_EXTENDED_STATS_JSON, "extended_stats.json", "InputFile.newStream"))
+                        .add(new FileOperation(TRINO_EXTENDED_STATS_JSON, "extended_stats.json", "OutputFile.createOrOverwrite"))
+                        .add(new FileOperation(TRANSACTION_LOG_JSON, "00000000000000000000.json", "InputFile.newStream"))
+                        .add(new FileOperation(TRANSACTION_LOG_JSON, "00000000000000000001.json", "InputFile.newStream"))
+                        .add(new FileOperation(TRANSACTION_LOG_JSON, "00000000000000000002.json", "InputFile.exists"))
+                        .add(new FileOperation(TRANSACTION_LOG_JSON, "00000000000000000002.json", "InputFile.newStream"))
+                        .add(new FileOperation(TRANSACTION_LOG_JSON, "00000000000000000002.json", "OutputFile.createOrOverwrite"))
+                        .add(new FileOperation(LAST_CHECKPOINT, "_last_checkpoint", "InputFile.newStream"))
+                        .add(new FileOperation(DATA, "no partition", "OutputFile.create"))
+                        .build());
+
         assertUpdate("DROP TABLE test_create_or_replace_as_select");
     }
 
+    @Test
+    public void testCreateOrReplaceTableWithCheckpoints()
+    {
+        assertUpdate("CREATE TABLE test_create_or_replace_with_checkpoints (id INT, data VARCHAR) WITH (checkpoint_interval = 2)");
+
+        assertUpdate(String.format("INSERT INTO test_create_or_replace_with_checkpoints VALUES (1, '2')"), 1);
+
+        assertUpdate("CREATE OR REPLACE TABLE test_create_or_replace_with_checkpoints " +
```

Comment:
here you should do `assertFileSystemAccesses` to verify that a checkpoint is actually being created and not when selecting from the table.